examples/external-data/src/mock-customer-service/service.ts

```diff
@@ -229,6 +273,64 @@ export async function initializeCustomerService(props: ServiceProps): Promise<Se
 		result.send();
 	});
 
+	/**
+	 * An 'events' endpoint that can be called by Fluid services.
+	 *
+	 * For the 'session-end' event {@link SessionEndEventsListenerRequest}: If, after unregistering the given client url, there are any task ids that have an outstanding
+	 * webhook registered using this services internal '/external-data-webhook' endpoint but has no respective active client sessions mapped to it anymore,
+	 * then that webhook will be deregistered.
+	 *
+	 * @remarks Currently, the only supported request type is 'session-end' {@link SessionEndEventsListenerRequest} which enables the Fluid service to notify this service
+	 * that a particular Fluid session has ended which in turn causes this service to unregister any related webhooks to the respective Fluid session.
+	 */
+	expressApp.post("/events-listener", (request: EventsListenerRequest, result) => {
+		const eventType = request.body?.type;
+
+		if (eventType === "session-end") {
+			const typedRequest = request as SessionEndEventsListenerRequest;
+			const containerUrl = typedRequest.body?.containerUrl;
+			if (containerUrl === undefined || typeof containerUrl !== "string") {
+				const errorMessage = `Missing or malformed containerUrl: ${containerUrl}`;
+				result.status(400).json({ message: errorMessage });
+				return;
+			}
+
+			// Removes the mapping of the given container URL from all task id's
+			const emptyTaskListRegistrationIds =
+				clientManager.removeAllClientTaskListRegistrations(containerUrl);
+			// If there are any task list id's that no longer have any active client sessions mapped to them
+			// then we should deregister our webhook for that task list id.
+			for (const emptyExternalTaskListId of emptyTaskListRegistrationIds) {
+				fetch(externalDataServiceWebhookUnregistrationUrl, {
+					method: "POST",
+					headers: {
+						"Access-Control-Allow-Origin": "*",
+						"Content-Type": "application/json",
+					},
+					body: JSON.stringify({
+						// External data service will call our webhook echoer to notify our subscribers of the data changes.
+						url: `http://localhost:${port}/external-data-webhook?externalTaskListId=${emptyExternalTaskListId}`,
+						emptyExternalTaskListId,
+					}),
```

Comment:
When unregistering a webhook in the external service it feels like the URL shouldn't be necessary anymore. During registration we provide both things because it needs to know where to issue a request to us when changes happen, but unregistering should only require the id, right? Does `externalDataServiceWebhookUnregistrationUrl` currently expect the URL?

Reply:
The external data service does expect the URL right now because its unsubscribing that webhook URL from changes for the given task list id. Ideally, we could have a bulk unsubscribe logic to unsub a given webhook from all task id's its currently registered to but right now that's not how it works

Reply:
Got it. Makes sense as is right now.